src/frontend/src/optimizer/plan_node/batch_simple_agg.rs

```diff
@@ -68,10 +70,45 @@ impl_plan_tree_node_for_unary! { BatchSimpleAgg }
 
 impl ToDistributedBatch for BatchSimpleAgg {
     fn to_distributed(&self) -> Result<PlanRef> {
-        let new_input = self
-            .input()
-            .to_distributed_with_required(Order::any(), &RequiredDist::single())?;
-        Ok(self.clone_with_input(new_input).into())
+        // Ensure input is distributed, batch phase might not distribute it
+        // (e.g. see distribution of BatchSeqScan::new vs BatchSeqScan::to_distributed)
+        let dist_input = self.input().to_distributed()?;
+
+        match dist_input.distribution() {
+            // 2-phase agg
+            Distribution::SomeShard => {
```

Comment:
correct me if I am wrong, if the input's distribution is `HashShard`, it can also be aggregated by two phases?

Reply:
yes, maybe we can change the condition like `if input.distribution().staisfy(Anyshard)`

Reply:
> correct me if I am wrong, if the input's distribution is HashShard, it can also be aggregated by two phases?
Won't a `LogicalAgg` with `HashShard` as input distribution always be converted into `BatchHashAgg` instead? Is there some example where `BatchSimpleAgg` could receive `HashShard` as input distribution?

Reply:
I guess I should treat the logic separate, and just look at semantics of batch simple agg. IF batch simple agg does receive hashshard, it is able to handle it too, with 2-phase aggregation.

Reply:
For example, `select sum(c1) from (select v1, sum(v2) as c1 from T group by v1)`;

The outer aggregation is a simple aggregation. The inner one is a hash aggregation and its output distribution obeys `HashShard(v1)`. This output is the input of the outer aggregation.

But it really depends on the implementation details whether the optimizer would still consider the input as `HashShard(v1)` after v1 gets `projected` away.